tools/wake-format/emitter.cpp

```diff
@@ -517,21 +613,25 @@ wcl::doc Emitter::walk_match(ctx_t ctx, CSTElement node) {
   MEMO(ctx, node);
   assert(node.id() == CST_MATCH);
 
-  MEMO_RET(
-      fmt()
-          .token(TOKEN_KW_MATCH)
-          .ws()
-          .walk(WALK(walk_node))
-          .consume_wsnl()
-          // clang-format off
-          .nest(fmt()
-              .fmt_while(
-                  CST_CASE, fmt()
-                  .newline()
-                  .walk(WALK(walk_node))
-                  .consume_wsnl()))
-          // clang-format on
-          .format(ctx, node.firstChildElement()));
+  MEMO_RET(fmt()
+               .token(TOKEN_KW_MATCH)
+               .ws()
+               .walk(WALK(walk_node))
+               // clang-format off
+               .nest(fmt()
+                   .consume_wsnl()
+                   .fmt_if(TOKEN_COMMENT, fmt().newline().token(TOKEN_COMMENT))
+                   .consume_wsnl()
```

Comment:
I think you can have white-space and newlines between multiple comments right? So this only handles one comment before each match, not multiple.

Reply:
ahh, that's actually a really good callout. Turns out this line isn't actually needed though and this is fully covered by the `fmt_while` just below. It was been removed

Reply:
Awesome, that's better.